internal/controller/pipelinegate_controller.go

```diff
@@ -152,64 +189,37 @@ func (r *PipelineGateReconciler) reconcilePendingGate(ctx context.Context, gate
 			return ctrl.Result{}, err
 		}
 
-		var gateState v1alpha1.GateState
-		if failed := hasFailed(reconciledJob); failed {
-			// if the job is failed, then we need to update the gate state to closed, unless it's a rerun
-			log.V(2).Info("Job failed.", "JobName", job.Name, "JobNamespace", job.Namespace)
-			gateState = v1alpha1.GateState(console.GateStateClosed)
-		} else if succeeded := hasSucceeded(reconciledJob); succeeded {
-			// if the job is complete, then we need to update the gate state to open, unless it's a rerun
-			log.V(1).Info("Job succeeded.", "JobName", job.Name, "JobNamespace", job.Namespace)
-			gateState = v1alpha1.GateState(console.GateStateOpen)
+		// ABORT:
+		if consoleclient.IsClosed(cachedGate) {
+			// I don't think a guarantee for aborting a job is possible, unless we change the console api to allow for it
+			// try to kill the job
+			err = killJob(ctx, r.Client, job)
+			// even if the killing of the job fails, we better update the gate status to closed asap, so we don't report a gate CR transition from pending to closed
+			gate.Status.SetState(console.GateStateClosed)
+			if err != nil {
+				log.Error(err, "Failed to kill Job.", "JobName", job.Name, "JobNamespace", job.Namespace)
+				return ctrl.Result{}, err
+			}
+			log.V(1).Info("Job aborted.", "JobName", job.Name, "JobNamespace", job.Namespace)
```

Comment:
similar here, just `return ctrl.Result{}, err` and reduce the need for indentation of the following